tests/test_functional_archive_logs.py

```diff
@@ -0,0 +1,200 @@
+
+    pytest.cicada_home = os.environ.get("CICADA_HOME")
+
+    pytest.db_host = os.environ.get("DB_POSTGRES_HOST")
+    pytest.db_port = os.environ.get("DB_POSTGRES_PORT")
+    pytest.db_user = os.environ.get("DB_POSTGRES_USER")
+    pytest.db_pass = os.environ.get("DB_POSTGRES_PASS")
+
+    pytest.db_test = f"pytest_{datetime.datetime.now().strftime('%Y%m%d_%H%M%S_%f')}"
+
+
+@pytest.fixture()
+def db_setup(get_env_vars):
+    """db_setup"""
+
+    # Create the test_db
+    pg_conn = psycopg2.connect(
+        host=pytest.db_host,
+        port=pytest.db_port,
+        user=pytest.db_user,
+        password=pytest.db_pass,
+        database="postgres",
+    )
+    pg_conn.autocommit = True
+    pg_cur = pg_conn.cursor()
+    pg_cur.execute(f"CREATE DATABASE {pytest.db_test}")
+
+    # Create test_db structure
+    test_conn = psycopg2.connect(
+        host=pytest.db_host,
+        port=pytest.db_port,
+        user=pytest.db_user,
+        password=pytest.db_pass,
+        database=pytest.db_test,
+    )
+    test_conn.autocommit = True
+    test_cur = test_conn.cursor()
+    test_cur.execute(
+        open(f"{pytest.cicada_home}/setup/schema.sql", "r", encoding="utf-8").read()
+    )
+    test_cur.close()
+    test_conn.close()
+
+
+def query_test_db(query):
+    """Run and SQL query in a postgres database"""
+    rows = []
+    with psycopg2.connect(
+        host=pytest.db_host,
+        port=pytest.db_port,
+        user=pytest.db_user,
+        password=pytest.db_pass,
+        database=pytest.db_test,
+    ) as conn:
+        conn.set_session(readonly=False, autocommit=True)
+        with conn.cursor() as cur:
+            cur.execute(query)
+            if cur.rowcount > 0 and cur.description:
+                rows = cur.fetchall()
+    return rows
+
+
+def test_test_db_setup(db_setup):
+    """test_test_db_setup"""
+    query_result = query_test_db("SELECT 1")
+
+    assert query_result == [(1,)]
+
+
+def test_create_dummy_servers():
+    """test_create_dummy_servers"""
+    query_test_db(
+        """
+        INSERT INTO servers
+            (server_id, hostname, fqdn, ip4_address, is_enabled)
+        VALUES
+            (1, '1', '1', '192.168.0.1', 1)
+        ;
+        """
+    )
+
+    results = query_test_db(f"SELECT count(*) FROM servers WHERE is_enabled=1")[0][0]
+
+    assert results == 1
+
+
+def test_create_dummy_schedules():
+    """test_create_dummy_schedules"""
+    query_test_db(
+        """
+        INSERT INTO schedules
+            (server_id, schedule_id, interval_mask, exec_command, is_enabled, is_running)
+        VALUES
+             (1, '1-1', '* * * * *', '1-1', 1, 0)
+            ,(1, '1-2', '* * * * *', '1-2', 1, 0)
+            ,(1, '1-3', '* * * * *', '1-3', 1, 1)
+            ,(1, '1-4', '* * * * *', '1-4', 1, 0)
+        ;
+        """
+    )
+
+    results = query_test_db(f"SELECT count(*) FROM schedules WHERE is_enabled=1")[0][0]
+
+    assert results == 4
+
+
+def test_create_dummy_schedule_logs():
```

Comment:
Why is this a test? this is more of a setup step that prepare data fixtures for main relevant test `test_archive_schedules`

Reply:
True, but having some check that the test data was inserted correctly has value.